lms/djangoapps/instructor_task/tasks_helper/utils.py

```diff
@@ -45,3 +54,41 @@ def tracker_emit(report_name):
     Emits a 'report.requested' event for the given report.
     """
     tracker.emit(REPORT_REQUESTED_EVENT_NAME, {"report_type": report_name, })
+
+
+class SensitiveMessageOnReports(object):
+    """
+    Check how to add the disclaimer new row in the CSV,
+    due to reports are generated of different ways, each one handles their own way:
+        1. Using the CSV library directly.
+        2. Using upload_csv_to_report_store function which in turn
+           uses store_rows of DjangoStorageReportStore class where is built the CSV.
+    """
+
+    def __init__(self):
+        self.display_msg = configuration_helpers.get_value('DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS', settings.FEATURES.get('DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS', False))
+
+    def csv_direct(self, writer):
+        """
+        Writes the row immediately in the CSV.
+        """
+        if self.display_msg:
+            msg = self.process_message()
+            encode = unicode(msg).encode('utf-8')
+            return writer.writerow([encode])
+
+    def with_report_store(self):
+        """
+        Return the string parsed in process_message. This string is passed
+        to upload_csv_to_report_store which decides if execute it or not.
+        """
+        return [self.process_message()]
+
+    def process_message(self):
+        """
+        Return the message parsed from template.
+        """
+        template_message = 'instructor/instructor_dashboard_2/sensitive_data_on_reports.txt'
```

Comment:
That's what I told you before. As I understand, they going to place the message in a template inside their theme. But here I'm calling a new template in the platform. So... I need to create this template in their theme and make a PR in that repo and replace this path by the created in the theme?

Reply:
I see now. My take on this is not for now. We do need to make sure that overriding the new template in the theme works, and that the string that is in the theme is used instead. Also we need to test that translating the string in the theme works.
Other than that, is out of the scope for now. We are assuming that the message is the same that they want, but if they wish to change it later on they must be able to at the theme.